### ISSUE TITLE
derivation subschema

### ISSUE THREAD

**Issue by dosumis** [HumanCellAtlas/metadata-schema#15]
This is a general ticket for discussion.  Specific implementation should be moved to subtickets. 

The V4 entity-relationship model for samples specifies that each sample should have fields to record which other sample (if any) a sample is derived from and what protocols were used in that derivation:

![image](https://user-images.githubusercontent.com/112839/31667687-150ad18e-b348-11e7-956b-13af5f908e1d.png)
![image](https://user-images.githubusercontent.com/112839/31667724-261bb54c-b348-11e7-8331-1591719015e6.png)

The version 3 schema has various fields for protocols used in derivation.  [sample.json](https://github.com/HumanCellAtlas/metadata-schema/blob/a01f82c62b799712e412f3740f970f76d1399f24/json_schema/sample.json)  has:

```js
        "disassociation_protocol": {
            "description": "How body part was treated to disassociate it into cells that can float individually in solution in sorter or microfluidics device.", 
            "pattern": "^P-....-.*$", 
            "type": "string"
      },
        "enrichment": {
            "description": "How cells were enriched for specific type or to weed out common types after disassociation.",
            "items": {
		"$ref": "http://hgwdev.soe.ucsc.edu/~kent/hca/schema/enrichment.json"
	    },
            "type": "array"
        }, 
        "preservation": {
            "$ref": "http://hgwdev.soe.ucsc.edu/~kent/hca/schema/preservation.json", 
            "description": "State of body_part at collection and how it was preserved after removal."
        }, 
        "protocol_ids": {
            "description": "Array of protocol IDs from project.protocols used for this sample.", 
            "items": {
                "pattern": "^.*-....-.*$", 
                "type": "string"
            }, 
            "type": "array"
        }
```

The two subschemas here capture considerable detail.  [enrichment.json](https://github.com/HumanCellAtlas/metadata-schema/blob/a01f82c62b799712e412f3740f970f76d1399f24/json_schema/enrichment.json) covers FACS markers; [preservation.json](https://github.com/HumanCellAtlas/metadata-schema/blob/a01f82c62b799712e412f3740f970f76d1399f24/json_schema/preservation.json) covers images of preserved samples.

TBD - how do we move from this model to one where we have standard dervied_from and derived_by (some protocol) fields for sample while preserving the ability to capture important detail and guide users towards which protocols and details are appropriate to record for specific sample types.

**Comment by dosumis** [comment]
CC @simonjupp @daniwelter @malloryfreeberg

**Comment by malloryfreeberg** [comment]
Following up on this ticket, I think most of these issues are being addressed in v5.0.0 with our Biomaterial -> Process/Protocol -> Biomaterial entity structure. We've added modules for things like enrichment, preservation/storage, dissociation, etc. that can hang off of the appropriate entities. E.g. the enrichment module is associated with the cell_suspension biomaterial.
